### PR TITLE
Mini compile

### PR DESCRIPTION
**Pull request by shayne-fletcher-da**
This PR provides an example of compiling a program to its `Core` representation.

### CODE REVIEW

File: examples/mini-compile/src/Main.hs [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+      createDirectoryIfMissing True (dataDir ++ "/../mingw")
+      cm <- runGhc (Just dataDir) $ do
+              setSessionDynFlags flags
+              compileToCoreSimplified file
+      putStrLn (showSDoc flags (ppr (cm)))
+    _ -> fail "Exactly one file argument required"
+
+mkDynFlags :: String -> String -> IO DynFlags
+mkDynFlags filename s = do {
+  ; dirs_to_clean <- newIORef Map.empty
```
Comment (neil-da) [inline examples/mini-compile/src/Main.hs]: SPJ style - grim - drop the `{;}` bits

File: examples/mini-compile/src/Main.hs [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+main :: IO ()
+main = do
+  args <- getArgs
+  case args of
+    [file] -> do
+      s <- readFile' file
+      flags <- mkDynFlags file s
+      dataDir <- getDataDir
+      createDirectoryIfMissing True (dataDir ++ "/../mingw")
```
Comment (neil-da) [inline examples/mini-compile/src/Main.hs]: Can we comment why this line is needed

File: examples/mini-compile/src/Main.hs [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+      cm <- runGhc (Just dataDir) $ do
+              setSessionDynFlags flags
+              compileToCoreSimplified file
+      putStrLn (showSDoc flags (ppr (cm)))
```
Comment (neil-da) [inline examples/mini-compile/src/Main.hs]: Run HLint over this - at least one redundant bracket

File: CI.hs [hunk 4]
```diff
@@ -34,7 +34,13 @@ main = do
     renameDirectory (dropExtension $ dropExtension tarball) "ghc-lib"
 
     -- Test the new projects
-    writeFile "stack.yaml" $ stackYaml ++ unlines ["- ghc-lib","- examples/mini-hlint"]
+    writeFile "stack.yaml" $
+      stackYaml ++
+      unlines ["- ghc-lib"
+              ,"- examples/mini-hlint"
+              , "- examples/mini-compile"
```
[the remark this answers is not in the document]
Comment (shayne-fletcher-da) [reply CI.hs]: ✔️

File: examples/mini-compile/src/Main.hs (hunk 1 above)
Comment (shayne-fletcher-da) [reply examples/mini-compile/src/Main.hs]: ✔️

File: examples/mini-compile/src/Main.hs (hunk 2 above)
Comment (shayne-fletcher-da) [reply examples/mini-compile/src/Main.hs]: ✔️

File: examples/mini-compile/src/Main.hs (hunk 3 above)
Comment (shayne-fletcher-da) [reply examples/mini-compile/src/Main.hs]: ✔️

**Pull request merged**